routers/repo/repo.go

```diff
@@ -268,6 +268,31 @@ func Action(ctx *context.Context) {
 	ctx.Redirect(redirectTo)
 }
 
+// RedirectDownload return a file based on the the following infos:
+func RedirectDownload(ctx *context.Context) {
+	var (
+		vTag     = ctx.Params("vTag")
+		fileName = ctx.Params("fileName")
+	)
+	tagNames := []string{vTag}
+	curRepo := ctx.Repo.Repository
+	releases, err := models.GetReleasesByRepoIDAndNames(curRepo.ID, tagNames)
+	if err != nil {
+		ctx.Handle(404, "RedirectDownload -> Release not found", err)
+		return
+	}
+
+	if len(releases) == 1 {
+		release := releases[0]
+		att, err := models.GetAttachmentByReleaseIDFileName(release.ID, fileName)
+		if err != nil {
+			ctx.Handle(404, "RedirectDownload -> Attachment not found", err)
```

Comment:
It will return ErrAttachmentNotExist error when no attachment by that filename exists. I think logic need to remade for GetAttachmentByReleaseIDFileName to return nil, nil when no attachment is found, so that it could be easily returned 404 on nil, nil and 500 on nil, err